backend/packages/identity-scan/src/scan/events/identity/identityTimelineOperations.js

```diff
@@ -0,0 +1,91 @@
+const {
+  identity: { getIdentityTimelineCollection },
+} = require("@statescan/mongo");
+const {
+  getCurrentBlockTimestamp,
+  getSubIdentityDisplay,
+} = require("../../utils/unitConversion");
+const {
+  EVENT_METHOD: {
+    JUDGEMENT_GIVEN,
+    JUDGEMENT_REQUESTED,
+    JUDGEMENT_UNREQUESTED,
+    SUB_IDENTITY_ADDED,
+    SUB_IDENTITY_REMOVED,
+    SUB_IDENTITY_REVOKED,
+  },
+} = require("../../constants");
+const { getIdentityStorage } = require("../../utils/getIdentityStorage");
+
+async function setIdentityEventForTimeline(method, event, indexer) {
+  let identityEvent = {};
+  const eventData = event.data;
+  const accountId = eventData[0].toString();
+
+  identityEvent = await getIdentityStorage(accountId);
+
+  //check if judgement or subidentity related method then add extra data
+  const judgementRelatedData = checkIfJudgementRelated(
+    method,
+    eventData,
+    identityEvent,
+  );
+  identityEvent = { ...identityEvent, ...judgementRelatedData };
+
+  const subIdentityRelatedData = await checkIfSubIdentityRelated(
+    method,
+    eventData,
+    identityEvent,
+  );
+  identityEvent = { ...identityEvent, ...subIdentityRelatedData };
+
+  identityEvent.method = method;
+  identityEvent.timestamp = await getCurrentBlockTimestamp(indexer);
```

Comment:
It's enough we save the indexer object to DB. We don't have to save the timestamp value. I suggest you check the implementation of assets-scan.